enhancements/sig-architecture/82-addon-template/README.md

```diff
@@ -0,0 +1,406 @@
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum:=KubeClient;CustomSigner
+	Type RegistrationType `json:"type"`
+
+	// KubeClient holds the configuration of the KubeClient type registration
+	// +optional
+	KubeClient *KubeClientRegistrationConfig `json:"kubeClient,omitempty"`
+
+	// CustomSigner holds the configuration of the CustomSigner type registration
+	// required when the Type is CustomSigner
+	CustomSigner *CustomSignerRegistrationConfig `json:"customSigner,omitempty"`
+}
+
+type KubeClientRegistrationConfig struct {
+	// Permissions represent the permission configurations of the addon agent to access the hub cluster
+	// +optional
+	Permissions []HubPermissionConfig `json:"permissions,omitempty"`
+}
+
+// HubPermissionConfig configures the permission of the addon agent to access the hub cluster.
+// Will create a RoleBinding in the same namespace as the managedClusterAddon to bind the user
+// provided ClusterRole/Role to the "system:open-cluster-management:cluster:<cluster-name>:addon:<addon-name>"
+// Group.
+type HubPermissionConfig struct {
+	// Type of the permissions setting. It defines how to bind the roleRef
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum:=CurrentCluster;SingleNamespace
+	Type HubPermissionsBindingType `json:"type"`
+
+	// RoleRef is an reference to the permission resource. it could be a role or a cluster role,
+	// the user must make sure it exist on the hub cluster.
+	// +kubebuilder:validation:Required
+	RoleRef rbacv1.RoleRef `json:"roleRef"`
+
+	// SingleNamespace contains the configuration of SingleNamespace type binding.
+	// It is required when the type is SingleNamespace
+	SingleNamespace *SingleNamespaceBindingConfig `json:"singleNamespace,omitempty"`
+}
+
+type SingleNamespaceBindingConfig struct {
+	// +kubebuilder:validation:Required
+	Namespace string `json:"namespace"`
+}
+
+type CustomSignerRegistrationConfig struct {
+	// signerName is the name of signer that addon agent will use to create csr.
+	// +required
+	// +kubebuilder:validation:MaxLength=571
+	// +kubebuilder:validation:MinLength=5
+	SignerName string `json:"signerName"`
+
+	// Subject is the user subject of the addon agent to be registered to the hub.
+	// If it is not set, the addon agent will have the default subject
+	// "subject": {
+	//	"user": "system:open-cluster-management:addon:{addonName}:{clusterName}:{agentName}",
+	//	"groups: ["system:open-cluster-management:addon", "system:open-cluster-management:addon:{addonName}", "system:authenticated"]
+	// }
+	Subject *Subject `json:"subject,omitempty"`
+
+	// SigningCA represents the reference of the secret to sign the CSR
+	// +kubebuilder:validation:Required
+	SigningCA SigningCARef `json:"signingCA"`
+}
+
+// SigningCARef is the reference to the signing CA secret that must contain the certificate authority data with key
+// "ca.crt"
+type SigningCARef struct {
+	// Namespace of the signing CA secret
```

Comment:
on the hub

Reply:
> on the hub
should clarify in the doc.   Probably also need to explain how the CSR is going to be approved.  I think we can know because the requestor will match the addon identity, but that's not spelled out.